src/services/api/rest/rest_api_server.cpp

```diff
@@ -269,7 +270,18 @@ void handleGetLogs(REQUEST_RESPONSE_PARAMS) {
         boost::property_tree::ptree pt;
         boost::property_tree::ptree list;
 
-        const std::vector<record_log_t>& logs = record_log_t::getLogs();
+        int limit = 100;
+        int beforeId = -1;
+        auto query_fields = request->parse_query_string();
+        for(auto &field : query_fields) {
+            if(field.first == LIMIT_QUERY_PARAM) {
+                limit = std::stoi(field.second);
+            } else if(field.first == BEFORE_ID_QUERY_PARAM) {
+                beforeId = std::stoi(field.second);
+            }
+        }
+
+        const std::vector<record_log_t>& logs = record_log_t::getLogs(limit, beforeId);
```

Comment:
For all web requests with pagination, we should include data in the response about what page the data is on. This could be something like `startIndex` and `endIndex`, or `pageStart` and `limit`.

Reply:
I'd prefer the syntax for the request to be `startIndex`/`pageStart` some other variation, where you specify a start index (_not_ a start ID), and the start index is relative to the pagination. So startIndex = 0 (default) would always return the most recent page.

I recognize that this might be really hard to do with the log entries since that would mean getting rid of the `where id < ...` clause and using `limit` and `offset` instead. We should test the performance out on a million or so rows. If the performance is <100ms, I think that's a better approach in terms of API design.